front/src/common/authorization/components/GrantsManagerSubjects.tsx

```diff
@@ -0,0 +1,140 @@
+import { ComboBox, Select } from '@osrd-project/ui-core';
+import cx from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import type { Grant, ResourceType, SubjectItemWithGrant } from 'common/api/mock/mockEditoastApi';
+import useSearchUser from 'common/useSearchUser';
+import { DEFAULT_GRANT } from 'modules/infra/consts';
+import generateGrantSelectProps from 'modules/infra/utils/generateGrantSelectProps';
+
+import useSubjectsResourceGrants from '../hooks/useSubjectsResourceGrants';
+
+type GrantsManagerSubjectsProps = {
+  resourceId: number;
+  resourceType: ResourceType;
+  userGrant: Grant;
+  resourceGrants?: {
+    [grant: string]: string[];
+  };
+};
+
+const GrantsManagerSubjects = ({
+  resourceId,
+  resourceType,
+  userGrant,
+  resourceGrants = {},
+}: GrantsManagerSubjectsProps) => {
+  const { t } = useTranslation('grantManagement');
+  const [displayUserSearchSection, setDisplayUserSearchSection] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<SubjectItemWithGrant[number] | undefined>();
+
+  const { userSubjectsList, updateUserResourceGrant } = useSubjectsResourceGrants({
+    resourceId,
+    resourceType,
+  });
+
+  const { searchResults, setSearchTerm, setSearchResults } = useSearchUser();
+
+  const resetSuggestions = () => {
+    setSearchResults([]);
+    setSearchTerm('');
+    setSelectedUser(undefined);
+  };
+
+  return (
+    <>
+      <div className="grant-manager-subject-search">
+        <button
+          type="button"
+          className="grant-manager-display-search"
+          onClick={() => setDisplayUserSearchSection(!displayUserSearchSection)}
+        >
+          {!displayUserSearchSection ? t('addGrantToUser') : t('translation:common.cancel')}
+        </button>
+        {displayUserSearchSection && (
+          <>
+            <div className="grant-manager-subject-search-section">
+              <span className="grant-manager-subject-search-combobox">
+                <ComboBox
+                  id="add-user-combobox"
+                  value={selectedUser}
+                  suggestions={searchResults}
+                  getSuggestionLabel={(option) => option.name}
+                  onChange={(e) => setSearchTerm(e.target.value)}
+                  onSelectSuggestion={(suggestion) => setSelectedUser(suggestion)}
+                  resetSuggestions={() => {}}
+                  autoComplete="off"
+                  narrow
+                />
+              </span>
+              <span className="grant-manager-subject-select">
+                <Select
+                  id="add-user-grant-selector"
+                  getOptionLabel={(option) => option.label}
+                  getOptionValue={(option) => option.value}
+                  onChange={(option) => {
+                    setSelectedUser((prev) =>
+                      prev
+                        ? {
+                            ...prev,
+                            grant: option?.value as Grant,
```

Comment:
Can you modify the type returned by `generateGrantSelectProps` so it returns now `{ label: string; value: Grant }[]` ? This way you can remove the cast
```suggestion
                            grant: option?.value,
```